File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/tools/TextTool.java
package ru.testsimpleapps.coloraudioplayer.managers.tools;


import android.graphics.Paint;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.widget.TextView;

public class TextTool {

    private static final String PATTERN_SHORT_NAME = "^[0-9]*|[a-zA-Z]{1,2}\\s*[0-9]*|[0-9]*\\s*[a-zA-Z]{1,2}$";
    private static final String PATTERN_NUMERIC = "^[0-9]*$";

    public static boolean isBadName(@NonNull final String number) {
        return (number != null && number.trim().matches(PATTERN_SHORT_NAME) ? true : false);
    }

    public static boolean isNumeric(@NonNull final String number) {
        return (number != null && number.trim().matches(PATTERN_NUMERIC) ? true : false);
    }

    public static int measureTextWidth(@NonNull final String text, @Nullable final TextView textView) {
        Paint paint = new Paint();
        if (textView != null) {
            paint = new  Paint(textView.getPaint());
        }

        return (int)paint.measureText(text);
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/tools/TextTool.kt
package ru.testsimpleapps.coloraudioplayer.managers.tools


import android.graphics.Paint
import android.widget.TextView

object TextTool {

    private val PATTERN_SHORT_NAME = "^[0-9]*|[a-zA-Z]{1,2}\\s*[0-9]*|[0-9]*\\s*[a-zA-Z]{1,2}$"
    private val PATTERN_NUMERIC = "^[0-9]*$"

    fun isBadName(number: String): Boolean {
        return if (number != null && number.trim { it <= ' ' }.matches(PATTERN_SHORT_NAME.toRegex())) true else false
    }

    fun isNumeric(number: String): Boolean {
        return if (number != null && number.trim { it <= ' ' }.matches(PATTERN_NUMERIC.toRegex())) true else false
    }

    fun measureTextWidth(text: String, textView: TextView?): Int {
        var paint = Paint()
        if (textView != null) {
            paint = Paint(textView.paint)
        }

        return paint.measureText(text).toInt()
    }

}
